sparsity/test/test_dask_sparse_frame.py

```diff
@@ -313,3 +325,50 @@ def test_from_ddf():
     with pytest.raises(ValueError):
         ddf = ddf.assign(A="some str value")
         dsf = dsp.from_ddf(ddf)
+
+
+def test_sdf_sort_index():
+    data = pd.DataFrame(np.random.rand(20, 4),
+                        columns=list('ABCD'),
+                        index=np.random.choice([1,2,3,4,5,6], 20))
+    ddf = dd.from_pandas(data,
+        npartitions=4,
+        sort=False,
+    )
+
+    dsf = dsp.from_ddf(ddf)
+    dsf = dsf.sort_index()
+
+    assert dsf.known_divisions
+
+    res = dsf.compute()
+    assert res.index.is_monotonic
+    assert res.columns.tolist() == list('ABCD')
+
+
+def test_sdf_sort_index_auto_partition():
+    data = pd.DataFrame(np.random.rand(20000, 4),
+                        columns=list('ABCD'),
+                        index=np.random.choice(list(range(5000)), 20000))
+    ddf = dd.from_pandas(data,
+        npartitions=20,
+        sort=False,
+    )
+
+    dsf = dsp.from_ddf(ddf)
+    dsf = dsf.sort_index(npartitions='auto', partition_size=80000)
+
+    assert dsf.known_divisions
+    assert dsf.npartitions == 16
```

Comment:
I don't know why this should be desired result. `partition_size` argument is not documented (I couldn't find documentation even in dask repo) - could you please describe this test a bit and maybe add at least `partition_size` description to `sort_index`'s docstring?